src/two_sum.rs

```diff
@@ -0,0 +1,69 @@
+use std::collections::HashMap;
+/*
+ *  I did not get to choose the signature here.  I believe it should return (usize, usize)...
+ *  or even Option<(usize, usize)>.
+ */
+pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
+    /*
+     *  Per the Constraints of the problem, there will only ever be exactly one solution.
+     *  incidentally, this means that either there will only be exactly _two_ of a number, or
+     *  that the number, doubled, does not equal the target.  In either case, I only then need
+     *  a single index for a given value, as during sum finding, it either won't matter, or
+     *  whichever one I _don't_ store will just be found pointing to the one I _did store.
+     */
+    let lookup: HashMap<i32, usize> = nums.iter().copied().enumerate().map(swap).collect();
+    let (idx, first) = nums
+        .into_iter()
+        .enumerate()
+        .find(|(i, x)| {
+            lookup
+                .get(&(target - x))
+                .copied()
+                /*
+                 *  I want there to be a better pattern for this?  Option implements Eq, but
+                 *  that's... not quite the same semantics
+                 */
+                .filter(|j| j != i)
+                .is_some()
+        })
+        .unwrap();
+    /*
+     *  Blech.  But the constraint is there _will_ be a solution, so the return
+     *  type does not allow for Monads
+     */
+
+    vec![idx, lookup.get(&(target - first)).copied().unwrap()]
```

Comment:
If all you need to do is assemble a few values so you have an iterator, it's more idiomatic to "just" use a [`slice`](https://doc.rust-lang.org/std/primitive.slice.html) -- it's a less involved and smaller allocation:

```rust
[idx, lookup.get(&(target - first)).copied().unwrap()].into_iter().map().collect()
```

Reply:
Didn't even know that was possible!